File: backend/entity/application_scholarship.go
package entity

import "gorm.io/gorm"

type ApplicationScholarship struct {
	gorm.Model
	Status string `json:"status"`

	ApplicationID uint        `json:"application_id"`
	Application   Application `gorm:"foreignKey:ApplicationID" json:"application"`

	ScholarshipID uint        `json:"scholarship_id"`
	Scholarship   Scholarship `gorm:"foreignKey:ScholarshipID" json:"scholarship"`

	ApplicationDocuments []ApplicationDocument `gorm:"foreignKey:ApplicationScholarshipID" json:"application_documents"`
	IntervieweBookings  []IntervieweBooking   `gorm:"foreignKey:ApplicationScholarshipID" json:"interviewe_bookings"`
}

File: frontend/src/interfaces/application_scholarship.ts
import type {
  ApplicationResponse,
  ScholarshipResponse,
  ApplicationDocumentResponse,
} from './';
import type { ScreeningResponse } from './screening';
import type { InterviewBooking } from './interview';
import type { EvaluationResponse } from './evaluation';

export interface ApplicationScholarshipResponse {
  ID: number;
  CreatedAt: string;
  UpdatedAt: string;
  status: string;

  application_id: number;
  application: ApplicationResponse;

  scholarship_id: number;
  scholarship: ScholarshipResponse;

  application_documents: ApplicationDocumentResponse[];
  screening?: ScreeningResponse;

  // Interview bookings
  interviewe_bookings?: InterviewBooking[];

  // Evaluation results
  evaluations?: EvaluationResponse[];
  final_total_score?: number;
  final_decision?: 'pending' | 'approved' | 'rejected' | 'waitlist';
}
